### PR TITLE
add customisation params to shake function

### PR DESCRIPTION
**Pull request by mhle**
ref: https://github.com/to-the-end/to-the-end/issues/104
allow customisation by passing in options to camera shake function. defaults are:
```
{
  shakeAxis: 'x', // valid options are 'x', 'y' and 'xy'
  shakeRange: 20,
  shakeInterval: 60,
  shakeCount: 10,
  randomShake: false,
  randomizeInterval: false
}
```

### CODE REVIEW

File: utils/camera.js [hunk 1]
```diff
@@ -1,7 +1,20 @@
 'use strict';
 
+const objectAssign = require('object-assign');
```
Comment (erbridge) [inline utils/camera.js]: We don't need this. Electron supports `Object.assign` out of the box.

**Comment by erbridge** [comment]
Sorry, I've set up linting, to ensure we have a consistent style. Run `npm run lint` to check your changes against it, and if anything rubs you the wrong way, feel free to submit a PR to change the rules.

File: utils/camera.js [hunk 2]
```diff
@@ -9,41 +22,57 @@ module.exports = {
     }
 
     const shakeTimer = game.time.create(false);
+    const shakeInterval = options.randomizeInterval? 
+      Math.random() * options.shakeInterval + options.shakeInterval : 
+      options.shakeInterval;
```
Comment (erbridge) [inline utils/camera.js]: Take a look at `Phaser.Math.integerInRange` for a more readable way to express this.

Comment (erbridge) [inline utils/camera.js]: Sorry, `game.rnd` not `game.math` (though you've used it below).

File: utils/camera.js [hunk 3]
```diff
@@ -9,41 +22,57 @@ module.exports = {
...
-        camera.x += shift;
+      if (options.shakeAxis === "x") {
+        game.camera.x += shift1;
+      }
 
-        shakeCount--;
+      if (options.shakeAxis === "xy") {
+        game.camera.x += shift1;
+        game.camera.y += shift2;
       }
```
Comment (erbridge) [inline utils/camera.js]: So we shake across the tl-br diagonal always if we aren't randomizing? I guess that's fine.

File: utils/camera.js [hunk 4]
```diff
@@ -9,41 +22,57 @@ module.exports = {
...
-            callback();
-          }
+      // Calculate camera shift
+      let shift1, shift2;
+      const shakeRangeHalved = options.shakeRange / 2;
 
-          return;
+      if (options.randomShake) {
+        shift1 = game.rnd.integerInRange(-shakeRangeHalved, shakeRangeHalved);
+        shift2 = game.rnd.integerInRange(-shakeRangeHalved, shakeRangeHalved);
```
Comment (erbridge) [inline utils/camera.js]: Can we call these `shiftX` and `shiftY` instead, since they're always used that way?

File: utils/camera.js [hunk 5]
```diff
@@ -9,41 +22,57 @@ module.exports = {
...
     const shakeTimer = game.time.create(false);
+    const shakeInterval = options.randomizeInterval? 
+      Math.random() * options.shakeInterval + options.shakeInterval : 
+      options.shakeInterval;
 
-    const shakeRange = 20;
-    const shakeInterval = 60;
-
-    let shakeCount = 10;
+    shakeTimer.loop(options.shakeInterval, () => {
```
Comment (erbridge) [inline utils/camera.js]: Could do this as a repeat instead of a loop.

File: utils/camera.js (hunk 1 above)
Comment (mhle) [inline utils/camera.js]: superweird, got an error first time I tried but now it works! will get rid of it

Comment (erbridge) [inline utils/camera.js]: Odd. I wonder why...

File: main.js [hunk 6]
```diff
@@ -9,7 +9,7 @@ const createWindow = function createWindow() {
     title:      'To The End',
     width:      1600,
     height:     900,
-    fullscreen: true,
+    fullscreen: false,
```
Comment (erbridge) [inline main.js]: Let's not commit this. I'll set up a PR that disables fullscreen when running `npm start`.

File: utils/camera.js [hunk 7]
```diff
@@ -9,40 +19,53 @@ module.exports = {
...
-          }
+        shakeTimer.stop();
+        return;
+      }
 
-          return;
-        }
+      // Calculate camera shift
+      let shift1;
+      let shift2;
```
Comment (erbridge) [inline utils/camera.js]: Rename these to `shiftX` and `shiftY`, since that's how they're used?

File: utils/camera.js (hunk 2 above)
Comment (mhle) [inline utils/camera.js]: Ok

File: utils/camera.js (hunk 4 above)
Comment (mhle) [inline utils/camera.js]: done

**Pull request merged**